packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -227,6 +231,30 @@ export const useArbTokenBridge = (
       childChainId: Number(l2NetworkID)
     })
 
+    addDepositToCache({
+      sender: walletAddress,
+      destination: walletAddress,
+      status: 'pending',
+      txID: tx.hash,
+      assetName: nativeCurrency.symbol,
+      assetType: AssetType.ETH,
+      l1NetworkID,
+      l2NetworkID,
+      value: utils.formatUnits(amount, nativeCurrency.decimals),
+      parentChainId: Number(l1NetworkID),
+      childChainId: Number(l2NetworkID),
+      direction: 'deposit',
+      type: 'deposit-l1',
+      source: subgraphExistsForChainPair({
+        parentChain: Number(l1NetworkID),
+        chain: Number(l2NetworkID)
+      })
+        ? 'subgraph'
+        : 'event_logs',
+      timestampCreated: String(dayjs().valueOf() / 1_000),
+      nonce: tx.nonce
```

Comment:
Requested by @fionnachan, adding nonce so that we can fix https://github.com/OffchainLabs/arbitrum-token-bridge/issues/986 in the future